public/js/graphing.js

```diff
@@ -0,0 +1,114 @@
+const canvas = document.getElementById("marks-graph");
+
+async function fetchStudentsMarks() {
+  
+  const url = server + '/marks'; //marks.json file
+  const options = {
+    method: 'GET',
+    headers: {
+      Accept: 'application/json',
+    },
+  };
+  const response = await fetch(url, options);
+  const studentsMarks = await response.json();
+  const dataMarks = studentsMarks; //fetched data goes to dataMarks
+  populateContent(studentsMarks);
+
+
+//generate the graph data 
+//processing the data fetched 
+let marks = [];
+
+for (let i = 0; i < dataMarks.length; i++) {
+  console.log(dataMarks[i].marks); //get the specific mark from the object
+  //marks = data[i].marks
+  marks.push(dataMarks[i].marks)
```

Comment:
```suggestion
  marks.push(parseInt(dataMarks[i].marks));
```

by making all the marks you push into the marks array integers you make sure that you're working with only one data type which avoids funky bugs like described in one of my comments in the review.